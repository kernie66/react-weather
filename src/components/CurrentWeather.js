export default function CurrentWeather() {
  return(
    <div class="d-flex justify-content-between ml-2">
    <div class="d-flex ml-2 mr-1">
        <img src="weather_icons/PNG/256/day_partial_cloud.png" width="75px" height="75px" alt="Väder"/>
    </div>
    <div class="d-flex flex-column ml-1 mr-1 pl-1 py-0">
        <div class="d-flex">
            <span class="outline-lg text-center">
                Växlande molnighet
            </span>
        </div>
        <div class="d-flex justify-content-center">
            <span class="outline-md text-center pb-2">
                Uppehåll
            </span>
        </div>
    </div>
</div>

  )
}