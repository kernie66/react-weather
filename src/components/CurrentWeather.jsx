import { Col, Container, Row } from "reactstrap";

export default function CurrentWeather() {
  return (
    <Container className="d-flex justify-content-around m-0">
      <Row>
        <Col xs="auto" className="p-0 m-0">
          <img src="weather_icons/PNG/256/day_partial_cloud.png" width="100px" height="100px" alt="Väder" />
        </Col>
        <Col xs="auto">
          <Row>
            <Col className="outline-lg text-center">
              Växlande molnighet
            </Col>
          </Row>
          <Row>
            <Col className="outline-md text-center fs-2 pb-2">
              Uppehåll
            </Col>
          </Row>
        </Col>
      </Row>
    </Container>

  )
}