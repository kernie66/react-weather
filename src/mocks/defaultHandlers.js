import { http, HttpResponse } from 'msw';

export const defaultHandlers = [
  http.get('*', () => {
    return HttpResponse.json({});
  }),
  http.post('*', () => {
    return HttpResponse.json({});
  }),
  http.put('*', () => {
    return HttpResponse.json({});
  }),
  http.patch('*', () => {
    return HttpResponse.json({});
  }),
  http.delete('*', () => {
    return HttpResponse.json({});
  }),
];
