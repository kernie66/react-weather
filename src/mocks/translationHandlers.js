import { http, HttpResponse } from 'msw';

const getWordCount = (text) => {
  return text ? text.trim().split(/\s+/g).length : 0;
};

const getMockedTranslations = (text) => {
  switch (text) {
    case 'Downpour':
      return 'Regnskur';
    case 'Expect a day of partly cloudy with rain':
      return 'RÃ¤kna med en dag med delvis molnigt med regn';
    case 'There will be partly cloudy today':
      return 'Det kommer att vara halvmolnigt idag';
    default:
      return 'Translated text: ' + getWordCount(text) + ' words';
  }
};
export const translationHandlers = [
  http.post(
    'https://translation.googleapis.com/language/translate/v2',
    async ({ request }) => {
      const body = await request.json();
      const translatedText = getMockedTranslations(body?.q);
      return HttpResponse.json({
        data: {
          translations: [
            {
              translatedText: translatedText,
              detectedSourceLanguage: 'en',
            },
          ],
        },
      });
    }
  ),
];
