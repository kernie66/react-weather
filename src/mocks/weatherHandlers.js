import { http, HttpResponse } from 'msw';
import weatherDataJson from '../../bruno/weather_alert_response.json';

const baseURL = import.meta.env.VITE_BASE_URL;

export const weatherHandlers = [
  http.get(`${baseURL}/onecall`, () => {
    // console.log('weatherDataJson', weatherDataJson);
    return HttpResponse.json(weatherDataJson);
  }),
];
