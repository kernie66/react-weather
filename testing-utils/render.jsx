// ./test-utils/render.tsx
import { render as testingLibraryRender } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
// Import your theme object
import { createTheme } from '@mantine/core';
import {
  QueryClient,
  QueryClientProvider,
} from '@tanstack/react-query';

const theme = createTheme({});

const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
    logger: {
      log: console.log,
      warn: console.warn,
      error: () => {},
    },
  });

export const testQueryClient = createTestQueryClient();

export function render(ui) {
  console.log('testQueryClient', testQueryClient);
  return testingLibraryRender(<>{ui}</>, {
    wrapper: ({ children }) => (
      <MantineProvider theme={theme}>
        <QueryClientProvider client={testQueryClient}>
          {children}
        </QueryClientProvider>
      </MantineProvider>
    ),
  });
}
