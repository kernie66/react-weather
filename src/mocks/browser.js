import { setupWorker } from 'msw/browser';
import { weatherHandlers } from './weatherHandlers.js';

export const worker = setupWorker(...weatherHandlers);
