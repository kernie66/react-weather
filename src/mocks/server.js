import { setupServer } from 'msw/node';
import { weatherHandlers } from './weatherHandlers.js';
import { defaultHandlers } from './defaultHandlers.js';
import { translationHandlers } from './translationHandlers.js';

export const server = setupServer(
  ...weatherHandlers,
  ...translationHandlers,
  ...defaultHandlers
);
