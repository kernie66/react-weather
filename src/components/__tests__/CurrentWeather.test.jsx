import { render, screen, userEvent } from '../../../testing-utils';
import CurrentWeather from '../CurrentWeather.jsx';
import { expect } from 'vitest';
import { http, HttpResponse } from 'msw';
import weatherRainDataJson from '../../../bruno/weather_rain_response.json';
import { testQueryClient } from '../../../testing-utils/render.jsx';
import { server } from '../../mocks/server';

const baseURL = import.meta.env.VITE_BASE_URL;

describe('test CurrentWeather', () => {
  beforeEach(() => {
    testQueryClient.removeQueries();
    localStorage.clear();
  });

  it('renders the current clear weather icon and text', async () => {
    const user = userEvent.setup();

    render(<CurrentWeather />);

    expect(
      screen.getByText(/väntar på väderdata/i)
    ).toBeInTheDocument();

    const weatherOverviewButton = await screen.findByRole('button', {
      name: /väderöversikt/i,
    });
    expect(weatherOverviewButton).toBeInTheDocument();
    expect(
      await screen.findByText(/växlande molnighet/i)
    ).toBeInTheDocument();
    expect(await screen.findByText(/uppehåll/i)).toBeInTheDocument();

    await user.click(weatherOverviewButton);
    expect(
      await screen.findByText(/väderöversikt för 2024-10-13/i)
    ).toBeInTheDocument();
    expect(
      await screen.findByText(/translated text 102/i)
    ).toBeInTheDocument();
  });

  it('renders the current rainy weather icon and text', async () => {
    server.use(
      http.get(`${baseURL}/onecall`, () => {
        console.log('weatherRainDataJson'); //, weatherRainDataJson);
        return HttpResponse.json(weatherRainDataJson);
      })
    );
    const user = userEvent.setup();

    render(<CurrentWeather />);

    expect(
      screen.getByText(/väntar på väderdata/i)
    ).toBeInTheDocument();

    const weatherOverviewButton = await screen.findByRole('button', {
      name: /väderöversikt/i,
    });
    expect(weatherOverviewButton).toBeInTheDocument();
    screen.debug();
    expect(await screen.findByText(/lätt regn/i)).toBeInTheDocument();
    expect(await screen.findByText(/0.3 mm\/h/i)).toBeInTheDocument();
    expect(await screen.findByText(/36%/i)).toBeInTheDocument();
    expect(
      await screen.queryByText(/uppehåll/i)
    ).not.toBeInTheDocument();

    await user.click(weatherOverviewButton);
    expect(
      await screen.findByText(/väderöversikt för 2024-10-13/i)
    ).toBeInTheDocument();
    expect(
      await screen.findByText(/translated text 102/i)
    ).toBeInTheDocument();
    screen.debug();
  });
});
