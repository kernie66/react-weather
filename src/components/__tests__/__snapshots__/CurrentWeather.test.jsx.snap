// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test CurrentWeather > renders the current clear weather icon and text 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <style
        data-mantine-styles="classes"
      >
        @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
      </style>
      <p
        class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
      >
        Väntar på väderdata
      </p>
    </div>
  </body>,
  "container": <div>
    <style
      data-mantine-styles="classes"
    >
      @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
    </style>
    <p
      class="mantine-focus-auto m_b6d8b162 mantine-Text-root"
    >
      Väntar på väderdata
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
